### PR TITLE
Add Github actions prototype pipeline

### PR DESCRIPTION
**Pull request by jaimergp**
## Description

Adding support for GH Actions CI.

## Todos
- [X] Implement feature / fix bug
- [ ] Add [tests](https://github.com/choderalab/openmmtools/tree/master/openmmtools/tests)
- [ ] Update [documentation](https://github.com/choderalab/openmmtools/tree/master/docs) as needed
- [ ] Update [changelog](https://github.com/choderalab/openmmtools/blob/master/docs/releasehistory.rst)Notable points that this PR has either accomplished or will accomplish.

## Status
- [ ] Ready to go

### CODE REVIEW

**Comment by jaimergp** [comment]
@jchodera, there you go!

I think you still need to manually disable AppVeyor on their dashboard. Windows builds will need fixing too (pip fails to build the C extensions).

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+jobs:
+  test:
+    name: Test on ${{ matrix.cfg.os }}, Python ${{ matrix.cfg.python-version }}, OpenMM ${{ matrix.cfg.openmm }}
+    runs-on: ${{ matrix.cfg.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          - { os: macOS-latest, python-version: 3.7, openmm: latest }
```
Comment (jchodera) [inline .github/workflows/CI.yml:24]: I think this should be python 3.6 and 3.7 right now , until OpenMM supports python 3.8!

**Comment by jaimergp** [comment]
Suggestions have been applied, @jchodera 👍 

If any other changes are needed before merge (see comments above for Windows), feel free to commit to this branch. Otherwise, we can have this merged now and work on that later.

**Comment by jaimergp** [comment]
Bump for attention :)

**Pull request merged**